### ISSUE TITLE
File-level CamelCase enum_value_naming causes compiler to fail

### ISSUE THREAD

**Issue by otto-dev** [scalapb/ScalaPB#676]
Thanks for #675 

I'm using the current version on the `master` branch (`v0.9.3`).

With `enum_value_naming: CAMEL_CASE` the compiler sometimes tries to reference the  `enum's` default value in its all-caps version, which does not exist. This seems to only happen if the enum is used inside of another `message`.

To reproduce:

```proto
syntax = "proto3";
import "scalapb/scalapb.proto";

option (scalapb.options) = {
  single_file: true
  enum_value_naming: CAMEL_CASE
};

enum ExampleEnum {
  UNDEFINED = 0;
  SOMETHING = 10;
}

message Example {
  ExampleEnum foo = 7;
}
```

Error:

```

[error] /foobar/messaging/target/scala-2.13/src_managed/main/TradingProto.scala:46:36: value UNDEFINED is not a member of object ExampleEnum
[error]     foo: ExampleEnum = ExampleEnum.UNDEFINED
[error]                                    ^
[error] /foobar/messaging/target/scala-2.13/src_managed/main/TradingProto.scala:55:36: value UNDEFINED is not a member of object ExampleEnum
[error]         if (__value != ExampleEnum.UNDEFINED) {
[error]                                    ^
[error] /foobar/messaging/target/scala-2.13/src_managed/main/TradingProto.scala:72:32: value UNDEFINED is not a member of object ExampleEnum
[error]         if (__v != ExampleEnum.UNDEFINED) {
[error]                                ^
[error] /foobar/messaging/target/scala-2.13/src_managed/main/TradingProto.scala:118:80: value UNDEFINED is not a member of object ExampleEnum
[error]       ExampleEnum.fromValue(__fieldsMap.getOrElse(__fields.get(0), ExampleEnum.UNDEFINED.javaValueDescriptor).asInstanceOf[_root_.com.google.protobuf.Descriptors.EnumValueDescriptor].getNumber)
[error]                                                                                ^
[error] /foobar/messaging/target/scala-2.13/src_managed/main/TradingProto.scala:125:169: value UNDEFINED is not a member of object ExampleEnum
[error]         ExampleEnum.fromValue(__fieldsMap.get(scalaDescriptor.findFieldByNumber(7).get).map(_.as[_root_.scalapb.descriptors.EnumValueDescriptor]).getOrElse(ExampleEnum.UNDEFINED.scalaValueDescriptor).number)
[error]                                                                                                                                                                         ^
[error] /foobar/messaging/target/scala-2.13/src_managed/main/TradingProto.scala:139:23: value UNDEFINED is not a member of object ExampleEnum
[error]     foo = ExampleEnum.UNDEFINED
[error]                       ^
[error] 6 errors found
[error] (Compile / compileIncremental) Compilation failed
```

It works if `ExampleEnum foo = 7;` is omitted.

**Comment by thesamet** [comment]
Oops! The [default value of the enum in our test case](https://github.com/scalapb/ScalaPB/blob/4d5cea5bdde4b181376707dd69e413cf6b60267a/e2e/src/main/protobuf/enum_options.proto#L14) (`Unknown`) was already in CamelCase, so it failed to catch that other generated code reference the default value directly, rather than through its Scala name. The last commit fixes that, and also adds the new option to our property-based testing, to ensure many different configurations are tested together.

**Comment by otto-dev** [comment]
>  The default value of the enum in our test case (Unknown) was already in CamelCase

That bug did everything it could to sneak through and survive. Thanks for fixing, again so fast! :rocket:

**Comment by thesamet** [comment]
Released in 0.9.4
